### PR TITLE
fix(package): use relpath to cwd for vcs dirtiness report

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

Address https://github.com/rust-lang/cargo/pull/14968#issuecomment-2555901072

> I think the ideal solution is to be relative to the current directory but that takes more work and this incremental improvement is great!

Sorry I should have noticed your comment earlier.

### CODE REVIEW

**Comment by epage** [comment]
> Sorry I should have noticed your comment earlier.

I intentionally didn't block on it or raise it as a concern.  This is my "ideal" but there are multiple points of view to be had and the risk of details that weren't blocking the PR.

I leave it to you if you'd want to move this forward.

**Review by epage — APPROVED** [review]

**Comment by weihanglo** [comment]
The working directory can be seen as either git's workdir, or user's current working directory. Since the dirtiness check is more a Git thing, this PR makes more sense to me.

```
[ERROR] 1 files in the working directory contain changes that were not yet committed into git:

../Cargo.toml

to proceed despite this and include the uncommitted changes, pass the `--allow-dirty` flag
```

**Pull request merged**
